rtengine/ipsoftlight.cc

```diff
@@ -0,0 +1,73 @@
+/* -*- C++ -*-
+ *
+ *  This file is part of RawTherapee.
+ *
+ *  Copyright 2018 Alberto Griggio <alberto.griggio@gmail.com>
+ *
+ *  RawTherapee is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  RawTherapee is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with RawTherapee.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#include "improcfun.h"
+
+namespace rtengine {
+
+void ImProcFunctions::softLight(float *red, float *green, float *blue, int istart, int jstart, int tW, int tH, int TS)
+{
+    if (!params->softlight.enabled || !params->softlight.strength) {
+        return;
+    }
+
+    const float blend = params->softlight.strength / 100.f;
+    const float orig = 1.f - blend;
+
+    const auto apply =
+        [=](float x) -> float
```

Comment:
This is a pleasure to read. :+1:

Only a nitpicking concerning line 39 above: Using `[=]` (capture all by copy) isn't recommended, better keep the scope minimal, i.e. `[blend, orig]` as far as I can see.